pilotage/api/views.py

```diff
@@ -0,0 +1,89 @@
+    DI_SERVICES = "di_services", "data·inclusion - Services"
+    DI_STRUCTURES = "di_structures", "data·inclusion - Structures"
+
+
+QUERIES = {
+    DataSetName.DI_SERVICES: {
+        "database": 2,
+        "type": "query",
+        "query": {
+            "source-table": 2052,
+            "fields": [
+                ["field", 59639, {"base-type": "type/Text"}],  # ID
+                ["field", 59644, {"base-type": "type/Text"}],  # Structure ID
+                ["field", 59647, {"base-type": "type/Text"}],  # Source
+                ["field", 59614, {"base-type": "type/Text"}],  # Nom
+                ["field", 59643, {"base-type": "type/*"}],  # Thematiques
+                ["field", 59627, {"base-type": "type/*"}],  # Frais
+                ["field", 59636, {"base-type": "type/*"}],  # Profils
+                ["field", 59649, {"base-type": "type/Text"}],  # Commune
+                ["field", 59623, {"base-type": "type/Text"}],  # Code Postal
+                ["field", 59654, {"base-type": "type/Text"}],  # Code Insee
+                ["field", 59615, {"base-type": "type/Text"}],  # Adresse
+                ["field", 59629, {"base-type": "type/Float"}],  # Longitude
+                ["field", 59651, {"base-type": "type/Float"}],  # Latitude
+                ["field", 59624, {"base-type": "type/*"}],  # Modes Accueil
+            ],
+            "filter": [
+                "starts-with",
+                ["field", 59654, {"base-type": "type/Text"}],
+                "000",  # Default to no-matchs
+                {"case-sensitive": False},
+            ],
+        },
+    },
+    DataSetName.DI_STRUCTURES: {
+        "database": 2,
+        "type": "query",
+        "query": {
+            "source-table": 2051,
+            "fields": [
+                ["field", 59591, {"base-type": "type/Text"}],  # ID
+                ["field", 59608, {"base-type": "type/Text"}],  # Commune
+                ["field", 59606, {"base-type": "type/Text"}],  # Code Postal
+                ["field", 59588, {"base-type": "type/Text"}],  # Code Insee
+                ["field", 59605, {"base-type": "type/Text"}],  # Adresse
+                ["field", 59594, {"base-type": "type/Float"}],  # Longitude
+                ["field", 59599, {"base-type": "type/Float"}],  # Latitude
+                ["field", 59590, {"base-type": "type/Text"}],  # Typologie
+            ],
+            "filter": [
+                "starts-with",
+                ["field", 59588, {"base-type": "type/Text"}],
+                "000",  # Default to no-matchs
+                {"case-sensitive": False},
+            ],
+        },
+    },
+}
+
+
+def dataset(request, name):
+    if name not in DataSetName:
+        return HttpResponseNotFound()
+
+    query = copy.deepcopy(QUERIES[DataSetName(name)])
+    if department := request.GET.get("department"):
+        max_length = 3 if department.startswith(("97", "98")) else 2
+        query["query"]["filter"][2] = str(department)[:max_length]
```

Comment:
C'est un peu magique le binding à la query ? Si c'est toujours sous la forme `query["query"]["filter"][2] = value` on pourrait bien utiliser un fonction utilitaire `bind_query_param(value)`

Sinon prochainement on aura besoin de rendre les paramètres explicites sur le query dict ? Si "never is better than right now", on pourrait garder comme ça et laisser un commentaire `TODO` avec une explication ?

Reply:
Oui je vais rajouter le TODO, et effectivement la prochaine amélioration c'est de créer la _query_ programmatiquement en mode _query builder_ plutôt que de l'avoir en dure.

Reply:
Une patterne builder me paraît logique aussi 👍